packages/astro/src/integrations/index.ts

```diff
@@ -136,6 +136,14 @@ export async function runHookConfigSetup({
 					}
 					addedClientDirectives.set(name, buildClientDirectiveEntrypoint(name, entrypoint));
 				},
+				addMiddleware: ({ order, entrypoint }) => {
+					if(typeof updatedSettings.middleware[order] === 'undefined') {
+						throw new Error(
+							`The "${integration.name}" integration is trying to add middleware but did not specify an order.`
+						);
+					}
```

Comment:
Hm, I don't like having a default when it's not obvious what the default should be. It really depends on the use-case; if you are doing something that modifies the response you want to be in `pre`, if you are adding locals then `post` is fine.

Reply:
I agree with Matthew on this one. Since the order is essential and we can't guess what the user intends to do, I think we need to force the user to choose.